File: ZenBlogClient/src/app/_models/blog.ts
import { CategoryDto } from "./category";
import { CommentDto } from "./commentDto";
import { UserDto } from "./userDto";

export class BlogDto{
  id;
  title;
  coverImage;
  blogImage;
  description;
  categoryId;
  category: CategoryDto;
  userId;
  createdAt;
  updatedAt;
  user: UserDto;
  comments: CommentDto[]




}

File: ZenBlogServer/ZenBlog.Domain/Entities/ZenBlogEntities/Blog.cs
using ZenBlog.Domain.Entities.Abstraction;
using ZenBlog.Domain.Entities.UserEntities;

namespace ZenBlog.Domain.Entities.ZenBlogEntities;

public sealed class Blog :BaseEntity
{
    public string Title { get; set; }
    public string CoverImage { get; set; }
    public string BlogImage { get; set; }
    public string Description { get; set; }
    public string CategoryId { get; set; }
    public Category Category { get; set; }
    public string UserId { get; set; }
    public  User User { get; set; }
    public ICollection<Comment> Comments { get; set; }
}
